packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -667,6 +671,25 @@ See documentation: ${createDocumentationLink({
     this.middleware.forEach(({ instance }) => {
       instance.started();
     });
+
+    // This is the automatic Insights middleware,
+    // added when `insights` is unset and the initial results possess `queryID`.
+    // Any user-provided middleware will be added later and override this one.
+    if (typeof this._insights === 'undefined') {
+      mainHelper.derivedHelpers[0].once('result', () => {
```

Comment:
it's good, `result` still gets called, but with `null` on the first derived helper. This code works :)